cluster-agent/utils/argocd_login_command.go

```diff
@@ -14,7 +14,12 @@ import (
 
 // This file is loosely based on the 'argocd login' CLI command (https://github.com/argoproj/argo-cd/blob/0a46d37fc6af9fe0aa963bdd845e3d799aa0320d/cmd/argocd/commands/login.go#L60)
 
-func generateDefaultClientForServerAddress(server string, optionalAuthToken string, skipTLSTest bool) (argocdclient.Client, error) {
+type tlsConfig struct {
+	skipTLSTest bool
+	testTLS     func(addr string, dialtime time.Duration) (*grpc_util.TLSTestResult, error)
+}
+
+func generateDefaultClientForServerAddress(server string, optionalAuthToken string, tlsCfg tlsConfig) (argocdclient.Client, error) {
```

Comment:
I have passed the `TestTLS` function as a dependency so that we can mock it in the unit tests. It's much easier than creating a fake gRPC server with TLS configurations. Let me know if this works and I'm open to suggestions if there is an easier way to test this.